lib/src/equality.dart

```diff
@@ -30,6 +30,38 @@ abstract class Equality<E> {
   bool isValidKey(Object o);
 }
 
+typedef F _GetKey<E, F>(E object);
+
+/// Equality of objects using a canonical value of the object itself.
+///
+/// For example, given the class:
+///     abstract class Employee {
+///       int get employmentId;
+///     }
+///
+/// Creating an [Equality] object that uses `employmentId`:
```

Comment:
Be a little more explicit. "The following [Equality] considers employees with the same IDs to be equal:"